src/blockchain/node/UpdateTransaction.cpp

```diff
@@ -44,17 +45,17 @@ auto UpdateTransaction::AddSibling(const block::Position& position) -> void
 {
     {
         auto& cached = siblings();
-        cached.emplace(position.second);
+        cached.emplace(position.hash_);
     }
 
-    add_sib_.emplace(position.second);
-    delete_sib_.erase(position.second);
+    add_sib_.emplace(position.hash_);
+    delete_sib_.erase(position.hash_);
 }
 
 auto UpdateTransaction::AddToBestChain(const block::Position& position) -> void
 {
-    RemoveSibling(position.second);
-    best_.emplace(position);
+    RemoveSibling(position.hash_);
+    best_.emplace(std::make_pair(position.height_, position.hash_));
```

Comment:
Not sure about that but aren't we actually losing the benefit of using `emplace` when we pass already constructed `std::pair`? In order to allow in-place construction we should pass the arguments and let `emplace` construct it itself. @wiesiekpap ?

Reply:
Yes, it's a bit nonsensical even if emplace invokes a move constructor. I'd be inclined to leave this as is in case it turns out someone is emotionally attached to emplace with move constructor.